client/my-sites/backup/backup-date-picker/test/index.jsx

```diff
@@ -56,13 +61,14 @@ describe( 'BackupDatePicker', () => {
 
 	test( "Shows 'Yesterday' as the previous date if the current date is today", () => {
 		const today = moment();
+		render( <BackupDatePicker selectedDate={ today } onDateChange={ () => {} } /> );
 
-		const picker = shallow( <BackupDatePicker selectedDate={ today } onDateChange={ () => {} } /> );
-
-		const previousDate = picker.find(
-			'.backup-date-picker__select-date--previous .backup-date-picker__display-date'
+		const yesterdayTextNode = screen.getByText( 'Yesterday' );
+		expect( yesterdayTextNode ).toBeInTheDocument();
+		expect( yesterdayTextNode ).toHaveClass( 'backup-date-picker__display-date' );
+		expect( yesterdayTextNode.parentElement ).toHaveClass(
+			'backup-date-picker__select-date--previous'
```

Comment:
Noting that tests that verify the DOM structure aren't very valuable when that structure is static. The only time when I see such tests as having value is when the `className` actually depends on a prop or any other internal logic.

This applies to multiple assertions in this PR, where the `className` is static and not changeable by any business logic.

Reply:
Hi @tyxla, I decided to do `className` because I wanted to make sure the date `yesterdayTextNode` appears only under the previous button in the date picker but not in any other place in the document(should not appear in the current date or next date). If I just have to check `toBeInTheDocument` the test might result in a false positive if it appears in any other place. And also there doesn't seem to be any role attached to the previous button in the date picker, so I can't query with `getByRole` to correctly assert that the date shows up only in the location i intended to.

Reply:
I understand, however, it's often the case that to properly assert that the structure is correct we often need a ton more  assertions. If we really want to do that, that's when snapshot tests come handy. Have you considered adding one in this scenario, rather than doing DOM traversing and checking for classnames?